pymc_bart/tree.py

```diff
@@ -52,66 +96,88 @@ class Tree:
         "output",
     )
 
-    def __init__(self, tree_structure, idx_leaf_nodes, output):
+    def __init__(
+        self,
+        tree_structure: Dict[int, Node],
+        idx_leaf_nodes: Optional[List[int]],
```

Comment:
Do we want this to be a list or a numpy array? In the `pgbart` module we have something like 

```python
        self.a_tree = Tree.new_tree(
            leaf_node_value=init_mean / self.m,
            idx_data_points=np.arange(self.num_observations, dtype="int32"),
            num_observations=self.num_observations,
            shape=self.shape,
        )
```
which makes mypy fail when using `List[int]`

Reply:
This generates the mypy error:
```python:
pymc_bart/pgbart.py:162: error: Argument "idx_data_points" to "new_tree" of "Tree" has incompatible type "ndarray[Any, dtype[Any]]"; expected "Optional[List[int]]"  [arg-type]
```

Reply:
a numpy array